### DISCUSSION TITLE
not able to get the cookies in middleware.js file in production, working fine in devlopment

### DISCUSSION THREAD

**Discussion by lokeshkumar0001** [vercel/next.js#64703] [App Router]
![image](https://github.com/vercel/next.js/assets/81354494/b4013cd7-135c-462e-8809-c44853c6e08a)

`import { NextResponse } from "next/server";
// import { cookies } from 'next/headers'
import { BASE_URL } from "./config/config";

const protectedRoutes = ["/dashboard","/form"];

export default async function middleware(req) {
  // const cookieStore = cookies();
  let isAuthenticated = false;

  const fetchUser = async () => {
    try {
      const accessToken = req.cookies.get('accessToken')?.value

      if(!accessToken){
        console.log('no accessToken found!')
      }
 
      if (accessToken) {
        const response = await fetch(`${BASE_URL}/api/user/me`, {
          method: "GET",
          headers: {
            Authorization: `Bearer ${accessToken}`
          }
        });

        if (response.ok) {
          const { success, user } = await response.json();
          if(success){
            
            isAuthenticated = true; // More concise check for successful user data
          }
        }
      }
    } catch (error) {
      console.error("Error fetching user:", error); // Log the error for debugging
    }
  };

  await fetchUser();

  if (!isAuthenticated) {
    const absoluteURL = new URL("/auth/login", req.nextUrl.origin);
    return NextResponse.redirect(absoluteURL.toString());
  }

  return NextResponse.next(); // Pass through the request if user is authenticated
}

export const config = {
  matcher: ['/about/:path*', '/dashboard/:path*','/form/:path*'],
}
`

Can any one me with this? I am not able to read the cookies in product although it is working in development

**Comment by SouthLink** [comment]
You can try to get cookies this way

```
export function middleware(request: NextRequest) {
  const cookies = request.cookies;

  console.log(cookies)

  return NextResponse.next();
}
```
